packages/next/src/build/index.ts

```diff
@@ -3029,7 +3029,10 @@ export default async function build(
           ),
           { overwrite: true }
         )
-        if (appDir) {
+        if (
+          appDir &&
+          (await fileExists(path.join(distDir, SERVER_DIRECTORY, 'app')))
+        ) {
```

Comment:
It's very strange that this happens, do you have a reproduction for this?

Reply:
I can create one if you want.

But that is because enabling `appDir: true` magically allows to `import { cache, use } from 'react'}`, even from pages directory.

Might not be the intended use, but feels better to add a `app/.gitkeep` than a dummy page.

Reply:
> But that is because enabling appDir: true magically allows to import { cache, use } from 'react'

Ah interesting! That sounds like a bug and might be fixed soon, enabling `app` shouldn't affect pages dir at all. I think we should remove this check in this PR then.